config/manager/manager.yaml

```diff
@@ -22,7 +22,7 @@ spec:
             - /manager
           args:
             - "--webhook-port=9443"
-            - "--enableBMHNameBasedPreallocation=${enableBMHNameBasedPreallocation:=false}"
+            - "--enableBMHNameBasedPreallocation=${ENABLE_BMH_NAME_BASED_PREALLOCATION:=false}"
```

Comment:
Q: Where is this env name coming? It is not present anywhere in Metal3 right now.

Reply:
OK, the issue links into discussion and it leads to https://github.com/kubernetes-sigs/cluster-api/pull/9513/files#diff-4127eae8106dc3a0709a4131a0007cc3dd8b0393ce07916829b1b4394603aecfR62 which seems also relevant as it says the upper case should be also used in clusterctl config. `scripts/ci-e2e.sh` L40 keeps the lowercase format though, should that also be fixed?

/cc @kashifest